src/Config.ts

```diff
@@ -134,245 +154,288 @@ export interface ConfigParams {
    * @category Number
    */
   thousandSeparator: '' | ',' | ' ' | '.',
   /**
-   * A list of time formats that are supported by time parsing functions.
+   * Sets time formats that will be supported by time-parsing functions.
    *
-   * The separator is ':' (colon).
+   * The separator is `:` (colon).
    *
-   * Any configuration of at least two of hh, mm, ss is accepted as a time, and they can be put in any order.
+   * Accepts any configuration of at least two of the following, in any order:
+   * - `hh`: hours
+   * - `mm`: minutes
+   * - `ss`: seconds
    *
    * @default ['hh:mm', 'hh:mm:ss.sss']
    *
    * @category Date and Time
    */
   timeFormats: string[],
   /**
-   * Specifies if the array arithmetic operations are allowed globally, or only inside special function (like ARRAYFORMULA).
+   * When set to `true`, array arithmetic is enabled globally.
+   * 
+   * When set to `false`, array arithmetic is enabled only inside array functions (`ARRAYFORMULA`, `FILTER`, and `ARRAY_CONSTRAIN`).
    *
+   * For more information, see the [Arrays guide](/guide/arrays.html).
+   * 
    * @default false
    *
    * @category Engine
    */
   useArrayArithmetic: boolean,
   /**
-   * Switches column search strategy from binary search to column index.
-   * Used by VLOOKUP and MATCH functions.
-   * Using column index may improve time efficiency but it will increase memory usage.
-   * In some scenarios column index may fall back to binary search despite this flag.
+   * When set to `true`, switches column search strategy from binary search to column index.
+   *
+   * Using column index improves efficiency of the `VLOOKUP` and `MATCH` functions, but increases memory usage.
+   * 
+   * When searching with wildcards or regular expressions, column search strategy falls back to binary search (even with `useColumnIndex` set to `true`).
    *
    * @default false
    *
    * @category Engine
    */
   useColumnIndex: boolean,
   /**
-   * Enables gathering engine statistics and timings. Useful for testing and benchmarking.
+   * When set to `true`, enables gathering engine statistics and timings.
+   * 
+   * Useful for testing and benchmarking.
    *
    * @default false
    *
    * @category Engine
    */
   useStats: boolean,
   /**
-   * A number of kept elements in undo history.
+   * Sets the number of elements kept in the undo history.
    *
    * @default 20
    *
    * @category Undo and Redo
    */
   undoLimit: number,
   /**
-   * If set true, then criterions in functions (SUMIF, COUNTIF, ...) can use regular expressions.
+   * When set to `true`, criteria in functions (SUMIF, COUNTIF, ...) are allowed to use regular expressions.
    *
    * @default false
    * @category String
    */
   useRegularExpressions: boolean,
   /**
-   * If set true, then criterions in functions (SUMIF, COUNTIF, ...) can use wildcards '*' and '?'.
+   * When set to `true`, criteria in functions (SUMIF, COUNTIF, ...) can use the `*` and `?`g wildcards.
```

Comment:
What does the `g` means here? Or is it a typo?
```suggestion
   * When set to `true`, criteria in functions (SUMIF, COUNTIF, ...) can use the `*` and `?` wildcards.
```

Reply:
@wojciechczerniak typo, fixed: https://github.com/handsontable/hyperformula/pull/743/files#diff-b83e1482492860acb73eb6a72535d0dd31b4fc971525ff13c35fb97222172d39R438